python/lsst/drp/tasks/reprocess_visit_image.py

```diff
@@ -0,0 +1,581 @@
+            before doing further processing.
+        initial_photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration that was applied to exposure during the
+            measurement of the background.
+        psf : `lsst.afw.detection.Psf`
+            PSF model for this exposure.
+        background : `lsst.afw.math.BackgroundList`
+            Total background that had been fit to the exposure so far;
+            modified in place to include background fit when detecting sources.
+        ap_corr : `lsst.afw.image.ApCorrMap`
+            Aperture Correction model for this exposure.
+        photo_calib : `lsst.afw.image.PhotoCalib`
+            Photometric calibration model for this exposure.
+        wcs : `lsst.afw.geom.SkyWcs`
+            World Coordinate System model for this exposure.
+        summary_stats : `lsst.afw.image.ExposureSummaryStats`
+            Summary statistics measured on this exposure.
+        calib_sources : `astropy.table.Table`
+            Per-visit catalog of measurements to get 'calib_*' flags from.
+        id_generator : `lsst.meas.base.IdGenerator`, optional
+            Object that generates source IDs and provides random seeds.
+        result : `lsst.pipe.base.Struct`, optional
+            Result struct that is modified to allow saving of partial outputs
+            for some failure conditions. If the task completes successfully,
+            this is also returned.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``exposure``
+                Calibrated exposure, with pixels in nJy units.
+                (`lsst.afw.image.Exposure`)
+            ``sources``
+                Sources that were measured on the exposure, with calibrated
+                fluxes and magnitudes. (`astropy.table.Table`)
+            ``sources_footprints``
+                Footprints of sources that were measured on the exposure.
+                (`lsst.afw.table.SourceCatalog`)
+            ``background``
+                Total background that was fit to, and subtracted from the
+                exposure when detecting ``sources``, in the same nJy units as
+                ``exposure``. (`lsst.afw.math.BackgroundList`)
+        """
+        if result is None:
+            result = pipeBase.Struct()
+        if id_generator is None:
+            id_generator = lsst.meas.base.IdGenerator()
+
+        result.exposure = self.snap_combine.run(exposures).exposure
+
+        # Calibrate the image, so it's on the same units as the background.
+        result.exposure.maskedImage = initial_photo_calib.calibrateImage(result.exposure.maskedImage)
+        result.exposure.maskedImage -= background.getImage()
+        # Uncalibrate so that we do the measurements in instFlux, because
+        # we don't have a way to identify measurements as being in nJy (TODO!).
```

Comment:
Another TODO?

Reply:
That's a very nebulous TODO, so I've removed it. This is vaguely related to DM-15762, but not quite the same. You can't ask an exposure "what units per pixel are you?", and I think we only have a concept of a plan for how to do so.